### PR TITLE
Create user and company

### PR DESCRIPTION
**Pull request by mwafrika**
# What does this pull do ?

- [ ] Create a user
- [ ] Create a company

### CODE REVIEW

**Comment by jefftrojan** [comment]
2 checks failed though, did you see that?

**Review by ciraganenicole — APPROVED** [review]

**Review by Irirwanirira — APPROVED** [review]

File: src/config/database.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-const dotenv = require("dotenv");
+const dotenv = require('dotenv');
 
 dotenv.config();
```
Comment (alainburindi) [inline src/config/database.js]: dotenv should be configured only once. In the app.js maybe

File: src/config/database.js [hunk 2]
```diff
@@ -18,8 +18,8 @@ module.exports = {
     username: process.env.TEST_DB_USERNAME,
     password: process.env.TEST_DB_PASSWORD,
     database: process.env.TEST_DB_NAME,
-    dialect: "postgres",
-    host: "localhost",
+    dialect: 'postgres',
+    host: 'localhost',
```
Comment (alainburindi) [inline src/config/database.js]: This one can also be put in the environment variables

File: src/controllers/auth.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+export const register = async (req, res) => {
+  const {
+    email,
+    password,
+    name,
+    companyName,
+    companyAddress,
+    companyPhone,
+    // roleId,
```
Comment (alainburindi) [inline src/controllers/auth.js]: Kindy remove commented codes

File: src/database/models/company.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+     * Helper method for defining associations.
+     * This method is not a part of Sequelize lifecycle.
+     * The `models/index` file will call this method automatically.
+     */
+    static associate(models) {
+      Company.hasMany(models.User, {
+        foreignKey: 'companyId',
+        as: 'users',
+        // onDelete: "CASCADE",
+        // onUpdate: "CASCADE",
```
Comment (alainburindi) [inline src/database/models/company.js]: Same here

File: src/utils/auth.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+};
+export const isTokenExpired = (token) => {
+  try {
+    const decoded = jwt.decode(token, { complete: true });
+    const currentTime = Math.floor(Date.now() / 1000); // in seconds
+    return decoded.payload.exp < currentTime;
+  } catch (error) {
+    return true;
+  }
+};
```
Comment (alainburindi) [inline src/utils/auth.js]: Not used so far

File: src/utils/format.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+export const formatResponse = (
+  res,
+  statusCode,
+  data = null,
+  errorMessage = null,
+  options = {},
+) => {
+  const successMessage = 'Success';
```
Comment (alainburindi) [inline src/utils/format.js]:
```suggestion
  const message = errorMessage || 'Success';
```

File: src/utils/format.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  res.status(statusCode);
+  const response = {
+    message: errorMessage || successMessage,
```
Comment (alainburindi) [inline src/utils/format.js]:
```suggestion
    message,
```

**Review by alainburindi — CHANGES_REQUESTED** [review]

**Review by alainburindi — APPROVED** [review]

**Pull request merged**